pkgs/development/python-modules/eth-hash/default.nix

```diff
@@ -1,52 +1,63 @@
 {
   lib,
-  fetchFromGitHub,
   buildPythonPackage,
+  fetchFromGitHub,
+  isPyPy,
   pythonAtLeast,
   pythonOlder,
+  pycryptodome,
   pytest,
+  pytest-xdist,
   safe-pysha3,
-  pycryptodome,
+  setuptools,
 }:
 
 buildPythonPackage rec {
   pname = "eth-hash";
-  version = "0.5.2";
-  format = "setuptools";
-  disabled = pythonOlder "3.5";
+  version = "0.7.0";
+
+  pyproject = true;
+
+  disabled = pythonOlder "3.8";
 
   src = fetchFromGitHub {
     owner = "ethereum";
     repo = "eth-hash";
-    rev = "v${version}";
-    hash = "sha256-6UN+kvLjjAtkmLgUaovjZC/6n3FZtXCwyXZH7ijQObU=";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-tFKq+WN8Z1BIAOIfaRtVt4+pnZ99FwHO8/pycmQx5Gg=";
   };
 
+  build-system = [ setuptools ];
+
   nativeCheckInputs =
-    [ pytest ]
+    [
+      pytest
+      pytest-xdist
+    ]
     ++ passthru.optional-dependencies.pycryptodome
-    # eth-hash can use either safe-pysha3 or pycryptodome;
-    # safe-pysha3 requires Python 3.9+ while pycryptodome does not.
-    # https://github.com/ethereum/eth-hash/issues/46#issuecomment-1314029211
-    ++ lib.optional (pythonAtLeast "3.9") passthru.optional-dependencies.pysha3;
+    # safe-pysha3 is not available on pypy
+    ++ lib.optional (!isPyPy) passthru.optional-dependencies.pysha3;
 
+  # Backends need to be tested separatly and can not use hook
```

Comment:
Are you sure? We can't set `pytestFlagsArray` depending on whether we're on PyPI? At least without pytest-xdist I'd hope it works.

Reply:
The problem is more that if both files are included, it causes duplicate test definition, as both backend test folders symlink to the same file, e.g. [here](https://github.com/ethereum/eth-hash/blob/main/tests/backends/pycryptodome/test_results.py).